metrics_utility/test/ccspv_reports/test_infrastructure_summary_broken_data.py

```diff
@@ -0,0 +1,441 @@
+            assert current_row is not None
+            # Verify that cells were written (indicating the method didn't crash)
+            assert mock_ws.cell.called
+        except Exception as e:
+            pytest.fail(f'Method failed with null/empty device types: {str(e)}')
+
+    def test_malformed_corrupted_input_data(self):
+        """Test malformed or corrupted input data."""
+        broken_df = self.create_broken_dataframe('malformed_facts')
+        dataframes = {
+            'job_host_summary': broken_df,
+            'main_jobevent': pd.DataFrame(),
+            'main_host': pd.DataFrame(),
+            'data_collection_status': pd.DataFrame(),
+        }
+
+        report = ReportCCSPv2(dataframes, self.extra_params)
+
+        mock_ws = self.create_mock_worksheet()
+
+        # Test that malformed JSON is handled gracefully
+        try:
+            current_row = report._build_data_section_infrastructure_summary(1, mock_ws, broken_df)
+            assert current_row is not None
+            # Should not crash on malformed JSON
+            assert mock_ws.cell.called
+        except Exception as e:
+            pytest.fail(f'Method failed with malformed data: {str(e)}')
+
+    def test_timezone_edge_cases_timestamps(self):
+        """Test timezone edge cases for first/last seen timestamps."""
+        broken_df = self.create_broken_dataframe('timezone_edge_cases')
+        dataframes = {
+            'job_host_summary': broken_df,
+            'main_jobevent': pd.DataFrame(),
+            'main_host': pd.DataFrame(),
+            'data_collection_status': pd.DataFrame(),
+        }
+
+        report = ReportCCSPv2(dataframes, self.extra_params)
+
+        mock_ws = self.create_mock_worksheet()
+
+        # Test timezone handling
+        try:
+            current_row = report._build_data_section_infrastructure_summary(1, mock_ws, broken_df)
+            assert current_row is not None
+            assert mock_ws.cell.called
+        except Exception as e:
+            pytest.fail(f'Method failed with timezone edge cases: {str(e)}')
+
+    def test_uuid_collision_scenarios(self):
+        """Test UUID collision scenarios."""
+        broken_df = self.create_broken_dataframe('uuid_collisions')
+        dataframes = {
+            'job_host_summary': broken_df,
+            'main_jobevent': pd.DataFrame(),
+            'main_host': pd.DataFrame(),
+            'data_collection_status': pd.DataFrame(),
+        }
+
+        report = ReportCCSPv2(dataframes, self.extra_params)
+
+        mock_ws = self.create_mock_worksheet()
+
+        # Test handling of duplicate host names with different remote IDs
+        try:
+            current_row = report._build_data_section_infrastructure_summary(1, mock_ws, broken_df)
+            assert current_row is not None
+            assert mock_ws.cell.called
+
+            # Verify that both records are processed despite having the same host name
```

Comment:
shouldn't this also test *how* they were processed?

Should the host end up being counted as both vmware and kubernetes and appear in two sheets?